### PR TITLE
Webpack: Add DllPlugin and DllReferencePlugin classes and properties

### PR DESCRIPTION
**Pull request by mikejhill**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/webpack/docs/wiki/list-of-plugins#dllplugin>>
- [x] Increase the version number in the header if appropriate.
  * _***No version number increment necessary**_
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.
  * _***tslint.json already exists**_

### CODE REVIEW

**Comment by hill0826** [comment]
Had the same problem with @dt-bot not being able to parse the header in #15192. Not sure what's the cause of this.

**Comment by yuit** [comment]
@hill0826  i wonder if it because you didn't add your name to the header. 🔔 @tkqubo @ mattlewis92 @bumbleblym @bcherny @tommytroylin

**Comment by bumbleblym** [comment]
might have something to do with #15058 @andy-ms

**Comment by andy-ms** [comment]
`dt-bot` is in need of an update. Just ignore it if tests pass.

File: webpack/index.d.ts [hunk 1]
```diff
@@ -689,6 +689,84 @@ declare namespace webpack {
...
+    class DllPlugin extends Plugin {
+        constructor(options: DllPlugin.Options | DllPlugin.Options[]);
+    }
+
+    namespace DllPlugin {
+        interface Options {
+            /**
+             * The absolute path to the manifest json file (output).
+             */
+            path: string,
```
Comment (bumbleblym) [inline webpack/index.d.ts]: @hill0826 could you
1. change the type member delimiters to `;` to match the rest of the file
2. add tests [here](https://github.com/hill0826/DefinitelyTyped/blob/e4566cbfc15637e31930a90a9bcaa75b8d25421e/webpack/webpack-tests.ts#L21-L29)

i'd also order these keys alphabetically.

**Review by bumbleblym — CHANGES_REQUESTED** [review]

File: webpack/index.d.ts (hunk 1 above)
Comment (hill0826) [inline webpack/index.d.ts]: Reformatted and re-organized type members.

Added tests as well. Can you review these? The gist of the test file as far as I could tell was to verify a usage of type declarations, but not to test against the actual Webpack implementation. I tried to add the tests in that format.

Comment (bumbleblym) [inline webpack/index.d.ts]: yup that's right. LGTM 👍

**Pull request merged**